torchinductor/codecache.py

```diff
@@ -39,10 +39,10 @@ def write(source_code, ext, extra=""):
         os.makedirs(subdir, exist_ok=True)
     path = os.path.join(subdir, f"{basename}.{ext}")
     if not os.path.exists(path):
-        # use a random temp file for thread safety
-        tmp_path = f"{path}.{random.randint(0, 2**31)}"
-        with open(tmp_path, "w") as fd:
-            fd.write(source_code)
+        # use a temp file for thread safety
+        fd, tmp_path = tempfile.mkstemp(dir=subdir)
```

Comment:
Not sure if its a concern, but mkstemp adds new restrictions here:

> The file is readable and writable only by the creating user ID. If the platform uses permission bits to indicate whether a file is executable, the file is executable by no one. The file descriptor is not inherited by child processes.

Reply:
More detail, I say this because in the ddp context of #1185 - we may run into permission errors, if say, the file is copied onto another machine and cannot be executed there. We may want a os.chmod here.

Reply:
We are writing this file to `/tmp/torchinductor_$USER` so other users should have their own directory.